library/x509_crt.c

```diff
@@ -2088,13 +2087,15 @@ static int x509_crt_find_parent_in(
             continue;
         }
 
-        *r_parent = parent;
-        *r_signature_is_good = signature_is_good;
-
         break;
     }
 
-    if (parent == NULL)
+    if( parent != NULL )
+    {
+        *r_parent = parent;
+        *r_signature_is_good = signature_is_good;
```

Comment:
There seems to be a problem here @d3zd3z , I think this is why CI in Zephyr is failing.
https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/48178/1/console

signature_is_good might be used un-initialized, cause the use of the variable is moved out of the for loop

Reply:
Seems, here, you are actually reverting this patch https://github.com/zephyrproject-rtos/mbedtls/commit/3437a1ba58ec1be23b12b86e86ef611d132db5c1
That's why the warning starting appearing